common/src/main/java/com/datatorrent/common/util/PubSubMessage.java

```diff
@@ -67,6 +70,11 @@ public static PubSubMessageType getPubSubMessageType(String identifier)
 
   }
 
+  public static String getScheme(DAG dag)
```

Comment:
What is the reason to implement this method on PubSubMessage class?

Reply:
Malhar operators are also using this functionality. It could be in any utility class related to PubSub. This class looked like a container for all PubSub protocol related classes.

Reply:
rebasing..

Reply:
I don't think that PubSubMessage is the right choice for such utility method, it does not use or depends on the scheme and message can be passed unchanged over secure and insecure connections. Please consider passing DAG to PubSubWebSocketClient where it can be used to construct proper connections.

Reply:
It can go to a different class, PubSubMessage looked like a convenient place to put it as I couldn't find a more appropriate class and there was another PubSub related inner class in there but now I see it is only message type.

I would rather it be in a common place that malhar can also use. I agree that it's probably better to make web socket client can handle the detail but it is internal to stram today and not used in malhar and don't want to perform a surgery to make PubSubWebSocketClient available and make changes across the board in malhar to use it now. I will put it in a separate class like PubSubProtocol.

Reply:
Malhar may have its own utility class if necessary. It will also help avoiding unnecessary version dependency of library/applications on the core. Additionally, once you are at it, can you add a constructor to SharedPubSubWebSocketClient that takes DAG as an argument and constructs it based on the attribute values?

Reply:
That's fine, I can remove the utility method. Let me look into the SharedPubSubWebSocketClient.